tests/experimental/datasources/test_spark_s3_datasource.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+import logging
+import os
+import re
+from typing import TYPE_CHECKING, List, cast
+
+import pandas as pd
+import pytest
+from moto import mock_s3
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.util import S3Url
+from great_expectations.experimental.datasources import SparkS3Datasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    S3DataConnector,
+)
+from great_expectations.experimental.datasources.file_path_data_asset import (
+    _FilePathDataAsset,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_s3_datasource import CSVAsset
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+
+logger = logging.getLogger(__file__)
+
+
+try:
+    import boto3
+except ImportError:
+    logger.debug("Unable to load boto3; install optional boto3 dependency for support.")
+    boto3 = None
+
+
+@pytest.fixture()
+def aws_region_name() -> str:
+    return "us-east-1"
+
+
+@pytest.fixture()
+def aws_s3_bucket_name() -> str:
+    return "test_bucket"
+
+
+@pytest.fixture(scope="function")
+def aws_credentials() -> None:
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+
+
+@pytest.fixture
+def s3(aws_credentials, aws_region_name: str) -> BaseClient:
```

Comment:
```suggestion
def s3_mock(aws_credentials, aws_region_name: str) -> BaseClient:
```

Reply:
Looks like this one hasn't been applied yet.

Reply:
@Kilo59 Good catch -- fixed now.  Thanks.